AutomatedTesting/Gem/PythonTests/Atom/TestSuite_Main_GPU.py

```diff
@@ -154,35 +153,19 @@ def wrap_run(self, request, workspace, editor, editor_test_results, launcher_pla
 
 
 @pytest.mark.parametrize("project", ["AutomatedTesting"])
-@pytest.mark.parametrize("launcher_platform", ['windows_generic'])
-class TestMaterialEditor(object):
-
-    @pytest.mark.parametrize("cfg_args,expected_lines", [
-        pytest.param("-rhi=dx12", ["Registering dx12 RHI"]),
-        pytest.param("-rhi=Vulkan", ["Registering vulkan RHI"])
-    ])
-    @pytest.mark.parametrize("exe_file_name", ["MaterialEditor"])
-    @pytest.mark.test_case_id("C30973986")  # Material Editor Launching in Dx12
-    @pytest.mark.test_case_id("C30973987")  # Material Editor Launching in Vulkan
-    def test_MaterialEditorLaunch_AllRHIOptionsSucceed(
-            self, request, workspace, project, launcher_platform, generic_launcher, exe_file_name, cfg_args,
-            expected_lines):
-        """
-        Tests each valid RHI option (Null RHI excluded) can be launched with the MaterialEditor.
-        Checks for the specific expected_lines messaging for each RHI type.
-        """
+@pytest.mark.parametrize("launcher_platform", ['windows_material_editor'])
+class TestMaterialEditor(EditorTestSuite):
+    # Remove -BatchMode from global_extra_cmdline_args since we need rendering for these tests.
+    global_extra_cmdline_args = []
+    use_null_renderer = False
+    material_editor_test = True
```

Comment:
Don't add a bool here and add special cases. 
What if we have another app to test? We would be creating bools everywhere and make this unmaintainable.

The RFC was about implementing a base TestSuite class where the EditorTestSuite extends and sets the test app to be editor.exe. Then for material editor test we would have MaterialEditorTestSuite that extends from the base test suite and sets the app to be MaterialEditor.exe.

Reply:
also in reply to your question about this being the final version, it is not, it was opened to help resolve a bug with `get_returncode()` which actually had to do with the `--runpythontest` command vs. the `--runpython` command which also wasn't working until this PR: https://github.com/o3de/o3de/pull/9136 

thank you for your feedback, will be updating it appropriately

Reply:
I updated this but there isn't a specific base class for suites yet, but that will be in the next update (as well as declaring the GPU test correctly)